src/figuresClasses.ts

```diff
@@ -2,10 +1,89 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+function getNumberWithTwoDecimalPlaces(number: number): number {
+  return Math.floor(number * 100) / 100;
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side cannot be 0 and less');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const sumOfOtherSides = [a, b, c]
+      .filter((side) => side !== longestSide)
+      .reduce((sum, side) => sum + side, 0);
+
+    if (longestSide >= sumOfOtherSides) {
+      throw new Error(
+        'The longest side of a triangle should be less than a sum of others',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = 0.5 * (this.a + this.b + this.c);
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return getNumberWithTwoDecimalPlaces(area);
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Circle implements Figure {
+  readonly shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius cannot be 0 and less');
```

Comment:
[CODE STYLE] - The error message should be more descriptive. For example, 'The radius of a circle must be a positive number.'